src/sql/src/plan/statement/ddl.rs

```diff
@@ -1528,6 +1529,82 @@ pub fn plan_create_views(
 ) -> Result<Plan, PlanError> {
     let source_desc = scx.get_item_by_resolved_name(&source_name)?.source_desc()?;
     match &source_desc.connection {
+        SourceConnection::LoadGenerator(LoadGeneratorSourceConnection { generator, .. }) => {
+            let names = generator.views();
+            if names.is_empty() {
+                sql_bail!(
+                    "cannot generate views from {:?} load generator sources",
+                    generator
+                );
+            }
+            let views = names
+                .into_iter()
+                .map(|(table, columns)| {
+                    let mut projection = vec![];
+                    for (i, (column_name, column_type)) in columns.iter().enumerate() {
```

Comment:
At one point I had the new `views()` fn return a `RelationDesc` instead of the `(string, string)` it currently does, but I couldn't find a way to convert `ScalarType` into a resolved data type (or into a string for resolving with `names::resolve`) needed by `Expr::Cast`.

Reply:
I'll let someone make the final call, but my take is that this isn't Kafka. No one is using this thing outside a getting started experience and being pragmatic is a resonable take.

Reply:
You can take a `ScalarType` to a `pgrepr::Type` and then take _that_ to a string. We're already doing the second half of this in `CREATE VIEWS` for PostgreSQL; would be good to wrap it up in a function and use it in both places!

https://github.com/MaterializeInc/materialize/blob/625c4888772d71a10c14b6b1f4e90eb24203a028/src/sql/src/plan/statement/ddl.rs#L1592-L1606